astroquery/dastcom5/core.py

```diff
@@ -0,0 +1,334 @@
+                list(ast_database.dtype.names[-4:-3]) +
+                list(ast_database.dtype.names[-2:])
+            ]
+        )
+        ast_database.rename(
+            columns={"ASTNAM": "NAME", "NO": "NUMBER", "CALEPO": "CALEPOCH"}, inplace=True
+        )
+        com_database = pd.DataFrame(
+            com_database[
+                list(com_database.dtype.names[:17]) +
+                list(com_database.dtype.names[-4:-3]) +
+                list(com_database.dtype.names[-2:])
+            ]
+        )
+        com_database.rename(
+            columns={"COMNAM": "NAME", "NO": "NUMBER", "CALEPO": "CALEPOCH"}, inplace=True
+        )
+        df = ast_database.append(com_database, ignore_index=True)
+        df[["NAME", "DESIG", "IREF"]] = df[["NAME", "DESIG", "IREF"]].apply(
+            lambda x: x.str.strip().str.decode("utf-8")
+        )
+        return df
+
+    def read_record(self, record):
+        """Read `DASTCOM5` record and return body data.
+
+        Body data consists of numpy array with custom dtype.
+
+        Parameters
+        ----------
+        record : int
+            Body record.
+
+        Returns
+        -------
+        body_data : numpy.ndarray
+            Body information.
+
+        """
+        ast_header, com_header = self.read_headers()
+        ast_path = os.path.join(self.dbs_path, "dast5_le.dat")
+        com_path = os.path.join(self.dbs_path, "dcom5_le.dat")
+        # ENDPT1 indicates end of numbered asteroids records
+        if record <= int(ast_header["ENDPT2"][0].item()):
+            # ENDPT2 indicates end of unnumbered asteroids records
+            if record <= int(ast_header["ENDPT1"][0].item()):
+                # phis_rec = record_size * (record_number - IBIAS - 1 (header record))
+                phis_rec = 835 * (record - ast_header["IBIAS0"][0].item() - 1)
+            else:
+                phis_rec = 835 * (record - ast_header["IBIAS1"][0].item() - 1)
+
+            with open(ast_path, "rb") as f:
+                f.seek(phis_rec, os.SEEK_SET)
+                body_data = np.fromfile(f, dtype=self.ast_dtype, count=1)
+        else:
+            phis_rec = 976 * (record - com_header["IBIAS2"][0].item() - 1)
+            with open(com_path, "rb") as f:
+                f.seek(phis_rec, os.SEEK_SET)
+                body_data = np.fromfile(f, dtype=self.com_dtype, count=1)
+        return body_data
+
+    def orbit_from_name(self, name):
+        """Return :py:class:`~astropy.table.Table` given a name.
+
+        Retrieve info from JPL DASTCOM5 database.
+
+        Parameters
+        ----------
+        name : str
+            NEO name.
+
+        Returns
+        -------
+        Table : ~astropy.table.Table
+            Near Earth Asteroid/Comet orbit parameters, all stacked.
+
+        """
+        records = self.record_from_name(name)
+        table = self.orbit_from_record(records[0])
+        for i in range(1, len(records)):
```

Comment:
why not just loop over records?  i.e., `for rec in records:`

Reply:
I did that for making a properly formatted table :sweat_smile: 

As, `orbit_from_record` returns a table with headers and all, I just wanted to append the remaining information rather than appending headers of every table. 
See this: https://github.com/astropy/astroquery/pull/1339/files#diff-31d1b0f544166f590f6f2f9fc2aa5916R253

Should I think of some better solution?

Reply:
Yes.  The code you've written here simply doesn't require indexing.  For example, this is exactly equivalent:
```
    for rec in records:
        table = vstack([table, self.orbit_from_record(rec)[1]])
```

Also, though, repeated `vstack`ing is inefficient.  I think something like:
```
orbits = [self.orbit_from_record(rec) for rec in records]
tbl = Table(orbits)
```
would have the same effect and be a lot more efficient.  You might need to do some more wrangling; I'm not sure at the moment what `orbit_from_record` returns

Reply:
Sure! Will do that!

Reply:
Fixed this @keflavich !